src/dmd/expressionsem.d

```diff
@@ -4013,6 +4039,41 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
     override void visit(CallExp exp)
     {
+        const old_shared_allowed = shared_allowed;
+        scope(exit)
+            shared_allowed = old_shared_allowed;
+
+        // @@@SHARED@@@
+        // we need to set allow shared for overload resolution
+        shared_allowed = true;
+        // after resolution is done
+        scope (exit)
```

Comment:
That's way too much code to put in a `scope (exit)`

Reply:
I have to ensure this runs before every return from here.
maybe later it can be cleaner.
I am under time-pressure.

Reply:
I've put it into it's own function nyow.